### DISCUSSION TITLE
Online tutorial - some snags

### DISCUSSION THREAD

**Discussion by lohancock** [NCAR/container-dtc-nwp#86] [Q&A]
Following the recent update to the containers, I tried out the tutorial from the beginning.

- One demur, I worked in a spot instance.  Just after selectdion c5.4xlarge, in the Configure Instance Details page, under Purchasing Option, i chose Request Spot Instances.  I do not think this mattered, did save some money.

- Second demur, I used Instance Connect instead of Putty.  Easier.  So for the Security Group, instead of MyIp I gave HTTP access to the Northern Virginia Instance Connect server, namely, 18.206.107.24/29..... And so I skipped instruction 4, setting up SSH access..... And instead of instruction 6, i just clicked the CONNECT button on the instances page with the new instance selected.  Note that it advises I choose the name root and I do and everything works fine with that.  (note that in a prelimniary round I tried using the name ec2-user but then I had to sudo things)....... I don't think using Instance Connect mattered either....

Then began working through the tutorial but there are anomalies.....

Almost all commands can be cut and pasted and they run, but there are exceptions:  the command    export PROJ_DIR=`pwd`   works if I type it in but not if I cut and paste.  I think it is some character that looks right but is the wrong unicode.  URL dialogue checks and fixes things like that but the command line is particular... 

same for the download.  It does not work via cut and paste but it does work if I type it all in.

Under the section on downloading data, the Sandy data download does not work but the others all do.  error is:  "curl (3) URL using bad/illegal format or missing URL

In every case, echo from the ls -ald command is different in details from what is set out to expect, such as presenting a different version number.

Everything else then works all the way to the cases.    Once there, I skipped the Sandy bd I had failed to download the data.  Went onward to Snow.

Under setup environment, once again  it is necessary to type in export PROJ_DIR=`pwd` rather than cut and paste.  Then it works. 

The command   PROJ_VERSION=4.0.0   does not work but I have tried out instead PROJ_VERSION="4.0.0" and no complaints

Under Run NWP Initialization components I run the Singularity command  singularity exec -B ${PROJ_DIR}/data/shapefiles:/home/data/shapefiles -B ${PROJ_DIR}/container-dtc-nwp/components/scripts/common:/home/scripts/common -B ${PROJ_DIR}/container-dtc-nwp/components/scripts/snow_20160123:/home/scripts/case -B ${CASE_DIR}/pythonprd:/home/pythonprd ../python_${PROJ_VERSION}.sif /home/scripts/common/run_python_domain.ksh  

but I get a fatal error:
FATAL:  could not open image /home/ec2-user/python_4.0.0.sif:  failed to retrieve path for /home/ec2-user/python_4.0.0.sif: lstat /home/ec2-user/python_4.0.0.sif: no such file or directory
 
then i stopped.

**Comment by mkavulich** [comment]
@lohancock Unfortunately our web setup through Drupal seems to have some problems with inserting "non-breaking space" characters seemingly randomly, and whether this produces errors when copy-pasted seems to be OS- and browser-dependent. Can you post the pages where you saw this error and I can try to fix the issue?

The singularity error you are seeing means you have not yet downloaded the python singularity image file, or it is in the wrong location. Have you gone through all the instructions for downloading the Singularity images? For example, the python page is https://dtcenter.org/nwp-containers-online-tutorial/software-containers/python-graphics, and the third section describes how to download the Python Singularity image.
